### PR TITLE
add sn30 pro+ firmware v3.01

### PR DESCRIPTION
**Pull request by sajattack**
This is my first time contributing here, hope I did everything right. If not let me know and I can fix it.

### CODE REVIEW

**Comment by sajattack** [comment]
I generated guid like this `appstream-util generate-guid "USB\VID_2DC8&PID_6002"` but `fwupdmgr get-devices` reports  a different guid already and when I try to install it says "No supported devices found"

**Comment by sajattack** [comment]
Do I need to put both dfu pid and regular pid in the <firmware> xml block or just dfu? Also fwupdmgr is reporting I have version 3279.82 installed, but 8bitdo reports this as version 3.02-beta5 so it says it's older.

**Comment by sajattack** [comment]
By the way, are you aware of this site with the firmware? http://download.8bitdo.com/Firmware/

**Comment by hughsie** [comment]
> but fwupdmgr get-devices reports a different guid

if you run fwupdmgr as root, it should display *how* it created the GUIDs, which should help.

> are you aware of this site with the firmware? http://download.8bitdo.com/Firmware/

Nope, and 8bitdo really should be uploading their firmware themselves onto the LVFS.

**Comment by sajattack** [comment]
Ok, it's showing the dfu mode one, that makes sense.

**Comment by superm1** [comment]
> Nope, and 8bitdo really should be uploading their firmware themselves onto the LVFS.

If that's too much for them, at least getting them to make some metadata and generate the CAB files as part of releasing new firmware would be a good start.

**Comment by hughsie** [comment]
Can you squash down the commits and force push to your branch please. Then I'll merge and upload to the LVFS.

**Comment by sajattack** [comment]
There's still one thing unresolved, fwupd is reporting I have version 3279.82 installed, when in fact it's version 3.02-beta5.

File: releases/SN30_Pro+/3.02-beta5/sn30pro+.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    <firmware type="flashed">78acf201-a624-5c0b-a521-e204be14e3d5</firmware>
+    <!-- USB\VID_2DC8&PID_5750 -->
+    <firmware type="flashed">b02f1319-c1c4-54f4-b7f2-5bb3552f6a9e</firmware>
+  </provides>
+  <url type="homepage">http://www.8bitdo.com/sn30pro-plus/</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>Proprietary</project_license>
+  <developer_name>8Bitdo</developer_name>
+  <releases>
+    <release urgency="medium" version="3.02-beta5" date="2019-11-17" timestamp="1574059883">
```
Comment (superm1) [inline releases/SN30_Pro+/3.02-beta5/sn30pro+.metainfo.xml]: this version string is going to break fwupd.  It should be in the format of "pair": https://fwupd.org/lvfs/docs/metainfo/version

Fwupd doesn't know how to represent the `-beta5`.  I think a better version to select may be 3.0195 to represent beta5.

File: releases/SN30_Pro+/3.0195/sn30pro+.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <firmware type="flashed">78acf201-a624-5c0b-a521-e204be14e3d5</firmware>
+    <!-- USB\VID_2DC8&PID_5750 -->
+    <firmware type="flashed">b02f1319-c1c4-54f4-b7f2-5bb3552f6a9e</firmware>
+  </provides>
+  <url type="homepage">http://www.8bitdo.com/sn30pro-plus/</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>Proprietary</project_license>
+  <developer_name>8Bitdo</developer_name>
+  <releases>
+    <release urgency="medium" version="3.0195" date="2019-11-17" timestamp="1574059883">
```
Comment (hughsie) [inline releases/SN30_Pro+/3.0195/sn30pro+.metainfo.xml]: That's not possible....

	tmp = g_strdup_printf ("%u.%02u", version / 100, version % 100);
	fu_device_set_version (FU_DEVICE (self), tmp, FWUPD_VERSION_FORMAT_PAIR);

the second number has to be < 100...

Comment (superm1) [reply releases/SN30_Pro+/3.0195/sn30pro+.metainfo.xml]: Well that's unfortunate.  We don't really have a good way to represent a beta release then.

**Comment by sajattack** [comment]
I guess I'll just upload 3.01. Not sure how I got beta5 anyway, and it bugged out the official firmware updater.

**Comment by hughsie** [comment]
> Not sure how I got beta5 anyway

Right, as a rule we don't put non-release versions on the LVFS anyway.

**Comment by sajattack** [comment]
This is ready for review.

**Pull request merged**